MMM-text-clock.js

```diff
@@ -173,69 +38,36 @@ Module.register('MMM-text-clock', {
       this.size = 'medium';
     }
 
-    const self = this;
+    this.getActiveWords = undefined;
+    this.letters = [];
+    this.wordMap = {};
 
-    setInterval(() => {
-      self.updateDom();
-    }, 10000);
-
-    self.updateDom();
+    this.sendSocketNotification('SET_LANGUAGE', this.language);
   },
 
-  getActiveWords: function (time) {
-    const wordIndexes = [];
-
-    wordIndexes.push(this.wordMap.it);
-    wordIndexes.push(this.wordMap.is);
-
-    const hours = time.getHours();
-    const minutes = time.getMinutes();
-
-    if ((minutes >= 3 && minutes <= 7) || (minutes >= 53 && minutes <= 57)) {
-      wordIndexes.push(this.wordMap.five);
-    } else if (
-      (minutes >= 8 && minutes <= 12) ||
-      (minutes >= 48 && minutes <= 52)
-    ) {
-      wordIndexes.push(this.wordMap.ten);
-    } else if (
-      (minutes >= 13 && minutes <= 17) ||
-      (minutes >= 43 && minutes <= 47)
-    ) {
-      wordIndexes.push(this.wordMap.a);
-      wordIndexes.push(this.wordMap.quarter);
-    } else if (
-      (minutes >= 18) & (minutes <= 22) ||
-      (minutes >= 38 && minutes <= 42)
-    ) {
-      wordIndexes.push(this.wordMap.twenty);
-    } else if (
-      (minutes >= 23 && minutes <= 27) ||
-      (minutes >= 33 && minutes <= 37)
-    ) {
-      wordIndexes.push(this.wordMap.twentyfive);
-    } else if (minutes >= 28 && minutes <= 32) {
-      wordIndexes.push(this.wordMap.half);
-    }
-
-    if (minutes >= 3 && minutes <= 32) {
-      wordIndexes.push(this.wordMap.past);
-    } else if (minutes >= 33 && minutes <= 57) {
-      wordIndexes.push(this.wordMap.to);
-    }
+  socketNotificationReceived: function (notification, payload) {
+    if (notification === 'SET_LANGUAGE') {
+      this.getActiveWords = payload.getActiveWords;
```

Comment:
Somehow `payload.getActiveWords` is `undefined`. I set out a question on the MagicMirror forum to ask why: https://forum.magicmirror.builders/topic/12871/how-to-send-a-function-as-payload-of-socket-notification